File: sdks/go/pkg/models/operations/getshieldinformationbarrierreportsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetShieldInformationBarrierReportsIDPathParams struct {
	ShieldInformationBarrierReportID string `pathParam:"style=simple,explode=false,name=shield_information_barrier_report_id"`
}

type GetShieldInformationBarrierReportsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetShieldInformationBarrierReportsIDPathParams
}

type GetShieldInformationBarrierReportsIDResponse struct {
	ClientError                    *shared.ClientError
	ContentType                    string
	ShieldInformationBarrierReport *shared.ShieldInformationBarrierReport
	StatusCode                     int64
}

File: ts-sdk/src/sdk/models/operations/getshieldinformationbarrierreportsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetShieldInformationBarrierReportsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=shield_information_barrier_report_id" })
  shieldInformationBarrierReportId: string;
}


export class GetShieldInformationBarrierReportsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetShieldInformationBarrierReportsIdPathParams;
}


export class GetShieldInformationBarrierReportsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  shieldInformationBarrierReport?: shared.ShieldInformationBarrierReport;

  @SpeakeasyMetadata()
  statusCode: number;
}
